src/library.js

```diff
@@ -883,6 +883,8 @@ LibraryManager.library = {
       ENV['PWD'] = '/';
       ENV['HOME'] = '/home/web_user';
       ENV['LANG'] = 'C.UTF-8';
+      // Browser language detection #8751
+      ENV['LANG'] = (typeof navigator === 'object' && navigator.languages && navigator.languages[0] || 'C').replace('-', '_') + '.UTF-8';
```

Comment:
How does the order of evaluation work here - is `&&` before `||` or the opposite? Might be clearer to add parens, I think.